bindings/matrix-sdk-ffi/src/client_builder.rs

```diff
@@ -19,18 +20,57 @@ use crate::{client::ClientSessionDelegate, error::ClientError, helpers::unwrap_o
 /// A list of bytes containing a certificate in DER or PEM form.
 pub type CertificateBytes = Vec<u8>;
 
-#[derive(Clone)]
-pub(crate) enum UrlScheme {
-    Http,
-    Https,
+#[derive(Debug, Clone)]
+enum HomeserverConfig {
+    Url(String),
+    ServerName(String),
+    ServerNameOrUrl(String),
+}
+
+#[derive(Debug, thiserror::Error, uniffi::Error)]
+#[uniffi(flat_error)]
+pub enum ClientBuildError {
+    #[error(transparent)]
+    Sdk(MatrixClientBuildError),
```

Comment:
I think that this should work as well:

```suggestion
    Sdk(#[from] MatrixClientBuildError),
```

Which avoids the boilerplate further bellow.